src/runtime_view_proxy.js

```diff
@@ -0,0 +1,170 @@
+    if (target * bpe >= maxArraySize || start * bpe >= maxArraySize || (end && end * bpe >= maxArraySize)) {
+      var len = end - start
+      var targetArray = getHeapBlock(type, target * bpe, len)
+      var sourceArray = getHeapBlock(type, start * bpe, len)
+      targetArray.set(sourceArray)
+      return heapBlock
+    } else {
+      return heapBlock.copyWithin(target, start, end)
+    }
+  },
+  setOverridden(array, offset) {
+    var offsetReal = (offset || 0) * heapBlock.BYTES_PER_ELEMENT
+    if (offsetReal >= maxArraySize || array.byteLength + offsetReal >= maxArraySize) {
+      var targetArray = getHeapBlock(type, offsetReal, array.length)
+      targetArray.set(array)
+    } else {
+      heapBlock.set(array, offset)
+    }
+  },
+  subarray(start, end) {
+    var startReal = (start || 0) * heapBlock.BYTES_PER_ELEMENT
+    var endReal = end ? end * heapBlock.BYTES_PER_ELEMENT : wasmMemory.byteLength
+    if (startReal >= maxArraySize || endReal >= maxArraySize) {
+      return getHeapBlock(type, startReal, endReal - startReal)
+    } else {
+      return heapBlock.subarray(start, end)
+    }
+  },
+  fill(value, start, end) {
+    var startReal = (start || 0) * heapBlock.BYTES_PER_ELEMENT
+    var endReal = end ? end * heapBlock.BYTES_PER_ELEMENT : wasmMemory.byteLength
+    if (startReal >= maxArraySize || endReal >= maxArraySize) {
+      const hb = getHeapBlock(type, startReal, endReal - startReal)
+      hb.fill(0, 0)
+      return heapBlock
+    } else {
+      return heapBlock.fill(value, start, end)
+    }
+  },
+  slice(start, end) {
+    var startReal = (start || 0) * heapBlock.BYTES_PER_ELEMENT
+    var endReal = end ? end * heapBlock.BYTES_PER_ELEMENT : wasmMemory.byteLength
+    if (startReal >= maxArraySize || endReal >= maxArraySize) {
+      const hb = getHeapBlock(type, startReal, endReal - startReal)
+      return hb.slice(start, end)
+    } else {
+      return heapBlock.slice(start, end)
+    }
+  },
+  get(target, property) {
+    if (typeof property === 'number' || typeof property === 'bigint') {
+      var memoryOffset = property * target.BYTES_PER_ELEMENT
+      if (memoryOffset >= maxArraySize) {
+        var heap = getHeapBlock(type, memoryOffset, 1);
+        return heap[0];
+      } else {
+        return this.heapBlock[property]
+      }
+    }
+
+    if (property === 'copyWithin') {
+      return this.copyWithin
+    }
+
+    if (property === 'set') {
+      return this.setOverridden
+    }
+
+    if (property === 'subarray') {
+      return this.subarray
+    }
+
+    if (property === 'fill') {
+      return this.fill
+    }
+
+    if (property === 'slice') {
+      return this.slice
+    }
+
+    return heapBlock[property]
+  },
+  set(target, property, value) {
+    if (typeof property === 'number' || typeof property === 'bigint') {
+      var memoryOffset = property * target.BYTES_PER_ELEMENT
+      if (memoryOffset >= maxArraySize) {
+        var heap = getHeapBlock(type, memoryOffset, 1);
```

Comment:
It seems like creating a new view for ever get/set might be a little excessive.. could we instead cache the views?   For single element get/set we could have fixed set of views, no?

Reply:
Yeah. I can make a set of N 4gb views